src/NeuralVerification.jl

```diff
@@ -100,4 +101,8 @@ const TOL = Ref(sqrt(eps()))
 set_tolerance(x::Real) = (TOL[] = x)
 export set_tolerance
 
+LazySets.overapproximate(set::Vector{<:HPolytope}, ::Type{Hyperrectangle}) = overapproximate(overapproximate.(set, Hyperrectangle), Hyperrectangle)
+LazySets.overapproximate(set::Vector{<:AbstractZonotope}, ::Type{Hyperrectangle}) = overapproximate(ConvexHullArray(set), Hyperrectangle)
+LazySets.overapproximate(set::Union{SymbolicIntervalMask, SymbolicIntervalGradient}, ::Type{Hyperrectangle}) = Hyperrectangle(low=low(set), high=high(set))
+
```

Comment:
These are type piracy (except the last one, where we own the type). Let's figure out a way to accomplish what these are trying to do without defining them this way. What is the major reason for the first two?

Reply:
Hmm okay - it's so that we can always use overapproximate(set, Hyperrectangle) in our get_bounds function in reachability.jl. I'm not sure the best way to keep that function generate without overloading overapproximate so that it can work for any oof our approximate reachable sets.

Reply:
Would it be better to create our own overapproximate function (with a different name) which we define for some types of inputs by calling to LazySets and for others with these implementations? we could call it "overapproximate_all" or something?

Reply:
Okay I just made our own reach_overapprox method in reachability.jl - let me know what you think, it at least resolves the piracy thing.